testutil/metrictest/assertions.go

```diff
@@ -0,0 +1,236 @@
+// Package metrictest is used for making assertions on metrics
+// produced by linkerd proxy.
+package metrictest
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/linkerd/linkerd2/testutil/metrictest/promm"
+)
+
+var (
+	addrRE      = regexp.MustCompile(`[0-9\.]+:[0-9]+`)
+	ipRE        = regexp.MustCompile(`[0-9\.]+`)
+	authorityRE = regexp.MustCompile(`[a-zA-Z\-]+\.[a-zA-Z\-]+\.svc\.cluster\.local:[0-9]+`)
+)
+
+// HasNoOutboundHTTPRequest returns error if there is any
+// series matching request_total{direction="outbound"}
+func HasNoOutboundHTTPRequest(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction": promm.Equals("outbound"),
+	})
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to run a check against the provided metrics: %w", err)
```

Reply:
Done in: https://github.com/linkerd/linkerd2/pull/8547/commits/c55ef6c2c236443f20aaf4936e51be805a83c2bb